docs/faq.md

```diff
@@ -119,6 +119,33 @@ Yes, it offers Envoy support in several ways:
 - Connect-gRPC has an [Envoy filter](https://github.com/connectrpc/envoy-demo)
 - Connect-Go supports gRPC health checking via https://github.com/connectrpc/grpchealth-go
 
+### My web request fails but the response looks valid. Why isn't it going through?
+
+Web requests that attempt to load resources from a domain outside the domain of the page
+require you to configure Cross-Origin Resource Sharing (CORS). Not doing so may cause protocol
+errors like "missing trailer", etc. If the response is valid and the request is to a different
+domain, make sure the server has the correct [CORS configuration](./cors.md#configurations-by-protocol).
+
+### Why does my Web client not receive the error code from my server?
+
+Most of the time, it's because the CORS setup of the server is incomplete. It's not enough to allow request origin, 
+methods, and headers—you also have to to expose response headers. See the [CORS documentation](./cors.md) for details.
+
+### How do I set a cookie on the server?
+
+Every service method on the server receives the `HandlerContext` as the second argument, which provides access to 
+response headers. You can set cookies with the `Set-Cookie` response header—for example:
+
+```
+ctx.responseHeader.append("Set-Cookie", "foo=bar; Max=Age=120")
+```
+
+### Why does the network explorer in the browser show strange characters in payload?
+
+The Protobuf binary format is efficient, but can't be rendered as text by the browser. You can switch to JSON with the 
+transport option `useBinaryFormat: false` to make troubleshooting easier. Unary RPCs use pure JSON payloads with the 
+Connect protocol.
```

Comment:
This answer is specific to Connect-ES clients, but we're in the section "The Connect protocol".

Users may also run into this issue with other browser clients, so it's valuable to have this information here, but we can be more specific, for example with:

```suggestion
The Protobuf binary format is efficient, but can't be rendered as text by the browser. If you use a Connect-ES client, you can switch to JSON with the [transport option](./web/choosing-a-protocol) `useBinaryFormat: false` to make troubleshooting easier. Unary RPCs use pure JSON payloads with the Connect protocol.
```